### PR TITLE
add a pr template

### PR DESCRIPTION
**Pull request by KendallDoesCoding**
Updated PR from #4299

### CODE REVIEW

**Comment by KendallDoesCoding** [comment]
Please review and get back to me on this.

**Review by ghost — APPROVED** [review]
Fine with me.

**Comment by ghost** [comment]
The template content seems fine to me, however I just wonder whether we really need it or not? Because Nodejs.org isn't the same as Nodejs application. Waiting for others' suggestions.

**Review by aymen94 — APPROVED** [review]

**Comment by ovflowd** [comment]
> The template content seems fine to me, however I just wonder whether we really need it or not? Because Nodejs.org isn't the same as Nodejs application. Waiting for others' suggestions.

I kinda agree here in the standpoint of this node being the main `Node` repo. But it still feels like, a PR template _might_ help first-time-contributors to follow a set of tasks.

A lot of people ignore the `CONTRIBUTING.md` files, so having some kind of text directly on their face, helps.

**Comment by ovflowd** [comment]
I'm proceeding with the merge here 🙂 (Once the author rebases the branch)

**Review by WaleedAshraf — COMMENTED** [review]
In my personal experience, I have seen that people start to ignore such checks or it adds more friction for the first-time contributor. Also, this is the only repo where we have such a PR template.

Maybe we can copy the general PR guidelines in this repo from here:
https://github.com/nodejs/node/blob/5fad0b93667ffc6e4def52996b9529ac99b26319/doc/contributing/pull-requests.md

Also, there is no issue with this specific request. Would it be better to add this after some discussion with contributors?

I'm not against it. I just felt that better to add such changes if there has been enough ask for it.
Regular contributors might be able to address it better than I can.

**Review by AugustinMauroy — APPROVED** [review]

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+## Type of change
+
+Please delete options that are not relevant.
+
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+- [ ] This change requires a documentation update
+
+# Proposed Changes
```
Comment (bnb) [inline .github/pull_request_template.md:13]: Please add a newline to separate headers, and use the correct heading level (only using one H1 per document).

```suggestion
## Proposed Changes

```

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+- change 1
+- change 2
+
+# Additional Info
```
Comment (bnb) [inline .github/pull_request_template.md:17]: Please add a newline to separate headers, and use the correct heading level (only using one H1 per document).

```suggestion
## Additional Info

```

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+- [ ] I have added tests that prove my fix is effective or that my feature works
+- [ ] New and existing unit tests pass locally with my changes
+- [ ] Any dependent changes have been merged and published in downstream modules
+- [ ] I have checked my code and corrected any misspellings
+
+# Screenshots
+
+Original | Updated
+:----------------------:|:-----------:
+** original screenshot  | ** updated screenshot **
```
Comment (bnb) [inline .github/pull_request_template.md:36]: Screenshots is probably not a section that's needed for nodejs/node.

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
+# Related Issue/Addition to code
+- Issue goes here
+
+## Type of change
+
+Please delete options that are not relevant.
```
Comment (bnb) [inline .github/pull_request_template.md:6]: This should probably be a comment:
```suggestion
<!-- Please delete options that are not relevant. -->
```

File: .github/pull_request_template.md (hunk 3 above)
Comment (nschonni) [reply .github/pull_request_template.md:36]: This is for the website, not nodejs/node

Comment (bnb) [reply .github/pull_request_template.md:36]: my bad, 100% thought I was in nodejs/node :)

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+- [ ] I have performed a self-review of my own code
+- [ ] I have commented my code, particularly in hard-to-understand areas
+- [ ] I have made corresponding changes to the documentation
+- [ ] My changes generate no new warnings
+- [ ] I have added tests that prove my fix is effective or that my feature works
+- [ ] New and existing unit tests pass locally with my changes
+- [ ] Any dependent changes have been merged and published in downstream modules
+- [ ] I have checked my code and corrected any misspellings
+
+# Screenshots
```
Comment (bnb) [inline .github/pull_request_template.md:32]: Please use correct headers:

```suggestion
## Screenshots
```

**Comment by bnb** [comment]
commented a few changes _should this get merged_.

**Comment by bnb** [comment]
@aymen94 why'd you merge without addressing requested changes?

**Comment by ovflowd** [comment]
This shouldn’t have been merged. @aymen94 could you please revert?

**Pull request merged**